src/anders_mathiesen_ex/ex05/myrand.py

```diff
@@ -0,0 +1,115 @@
+# -*- coding: utf-8 -*-
+
+__author__ = 'Anders Mathiesen'
+__email__ = 'andermat@nmbu.no'
+
+
+class RandIter:
+    def __init__(self, random_number_generator, length):
+        """
+        Arguments
+        ---------
+        random_number_generator :
+            A random number generator with a ``rand`` method that
+            takes no arguments and returns a random number.
+        length : int
+            The number of random numbers to generate
+        """
+        self.generator = random_number_generator
+        self.length = length
+        self.num_generated_numbers = 0
+        self.is_iter = False
+
+    def __iter__(self):
+        """
+        Initialise the iterator.
+
+        Returns
+        -------
+        self : RandIter
+
+        Raises
+        ------
+        RuntimeError
+            If iter is called twice on the same RandIter object.
+        """
+        if self.is_iter:
+            raise RuntimeError
```

Comment:
The `RuntimeError` should include an explanation of what went wrong. This can be done by passing a string as argument to the exception:  `raise RuntimeError('what went wrong')`